src/persist/src/indexed/mod.rs

```diff
@@ -292,26 +355,203 @@ impl<U: Buffer, L: Blob> Indexed<U, L> {
 
         self.graveyard.push(mapping);
 
-        self.try_set_meta(prev)?;
+        self.try_set_meta()?;
+
         Ok(true)
     }
 
+    /// Validates the following preconditions for draining pending requests:
+    ///
+    /// - The meta we might roll back to must be equivalent to the durably
```

Comment:
you know how in middle school and high school teachers would tell you to not repeat the same word many times in an essay because thats somehow better writing? i feel like i subconsciously hit that when writing docs sometimes :)